client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -232,20 +233,31 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerI
     public AsyncIterator<Stream> listStreams(String scopeName) {
         Exceptions.checkNotClosed(closed.get(), this);
         long traceId = LoggerHelpers.traceEnter(log, "listStreams", scopeName);
-
+        long requestId = requestIdGenerator.get();
         try {
             final Function<ContinuationToken, CompletableFuture<Map.Entry<ContinuationToken, Collection<Stream>>>> function =
                     token -> this.retryConfig.runAsync(() -> {
                         RPCAsyncCallback<StreamsInScopeResponse> callback = new RPCAsyncCallback<>(traceId, "listStreams");
                         ScopeInfo scopeInfo = ScopeInfo.newBuilder().setScope(scopeName).build();
-                        new ControllerClientTagger(client).withTag(traceId, "listStreams", scopeName)
+                        new ControllerClientTagger(client).withTag(requestId, "listStreams", scopeName)
                                                           .listStreamsInScope(StreamsInScopeRequest
                                                                   .newBuilder().setScope(scopeInfo).setContinuationToken(token).build(), callback);
                         return callback.getFuture()
                                        .thenApply(x -> {
-                                           List<Stream> result = x.getStreamsList().stream()
-                                                                  .map(y -> new StreamImpl(y.getScope(), y.getStream())).collect(Collectors.toList());
-                                           return new AbstractMap.SimpleEntry<>(x.getContinuationToken(), result);
+                                           switch (x.getStatus()) {
+                                               case SCOPE_NOT_FOUND:
+                                                   log.warn(requestId, "Scope not found: {}", scopeName);
+                                                   throw new ScopeDoesNotExistException();
+                                               case FAILURE:
+                                                   log.warn(requestId, "Internal Server Error while trying to list streams in scope: {}", scopeName);
+                                                   throw new RuntimeException("Failure while trying to list streams");
```

Comment:
Is there anything else we can get to indicate the cause of the failure?

Reply:
This is placeholder enumeration for generic exceptions where we will explicitly fail the request. 
Currently for this api, that response code is never actually sent from the server

Reply:
Actually we will keep adding specific error codes to the enum just like we have SCOPE_NOT_FOUND currently. 
So FAILURE is generic. 
And as i said earlier, it isnt sent from the server as of now. So its only a placeholder.